cdc/model/schema_storage.go

```diff
@@ -88,12 +108,22 @@ func WrapTableInfo(schemaID int64, schemaName string, version uint64, info *mode
 
 	for i, idx := range ti.Indices {
 		ti.indicesOffset[idx.ID] = i
-		if ti.IsIndexUnique(idx) {
+		if ti.IsIndexUnique(idx, false) {
 			for _, col := range idx.Columns {
 				ti.uniqueColumns[ti.Columns[col.Offset].ID] = struct{}{}
 			}
 		}
-		if idx.Primary || idx.Unique {
+		if ti.IsIndexUnique(idx, true) {
```

Comment:
suggest to use another function for the condition of `allowColumnNullable=true`

Reply:
fine